src/material/table/_table-flex-styles.scss

```diff
@@ -0,0 +1,72 @@
+/**
+ * Flex-based table structure
+ */
+$mat-header-row-height: 56px;
+$mat-row-height: 48px;
+$mat-row-horizontal-padding: 24px;
+
+// Only use tag name selectors here since the styles are shared between MDC and non-MDC
+@mixin mat-private-table-flex-styles {
```

Comment:
Looking through this again, do you think that we should make these selectors a bit more specific? Currently `mat-table {}` will match both the MDC and non-MDC elements which could interfere with each other. We could change it to something like `mat-table.mat-mdc-table`, but I'm not familiar enough to know whether it'll make much of a difference.

Reply:
The reason for extracting the flex styles into a mixin is so that both MDC and non-MDC table can use the styles

Reply:
Wouldn't we want the MDC version to be using MDC's styles though? Otherwise it's basically a duplicate of our current table.

Reply:
MDC only has native table styles. The flexbox-based table is an Angular special.

Reply:
Wouldn't we still want to make it work using MDC's styles? Not necessarily in this PR, but we can work with the MDC team to split their styles up into more granular mixins so we can apply it to our flexbox table too.

Reply:
I had been assuming the only styles we were applying exclusively to the flex table were the actual flexbox styles. If there are styles common between them, we should just use the appropriate css classes for those.